### PR TITLE
Improve memory footprint

### PR DESCRIPTION
**Pull request by banesullivan**
Duplicate of #327 - I've made a few changes

Also, I've added [`pytest-memprof`](https://pypi.org/project/pytest-memprof/) to the dependencies during testing so that stats are automatically reported for high consupmption tests. Here's what I have from running locally:

```
============================================= memory consumption estimates =============================================
tests/test_plotting.py::test_plot_multi_component_array            - 482.6 MB
tests/test_plotting.py::test_plot_show_bounds_params[origin-both]  - 272.9 MB
tests/test_plotting.py::test_plot_show_bounds_params[back-back]    - 233.9 MB
tests/test_plotting.py::test_closing_and_mem_cleanup               - 232.3 MB
tests/test_plotting.py::test_plot_show_bounds_params[front-True]   - 220.3 MB
tests/test_plotting.py::test_volume_rendering                      - 215.8 MB
tests/test_plotting.py::test_plot_label_fmt                        - 175.6 MB
tests/test_ipy_tools.py::test_ipy_integrated                       - 107.1 MB
tests/test_examples.py::test_docexample_advancedplottingwithnumpy  - 98.3 MB
tests/test_plotting.py::test_scalars_by_name                       - 98.0 MB
tests/test_examples.py::test_beam_example                          - 82.1 MB
tests/test_grid.py::test_save_rectilinear[vtk-True]                - 49.3 MB
tests/test_polydata.py::test_boolean_difference                    - 18.9 MB
tests/test_examples.py::test_plot_ants_plane                       - 18.5 MB
tests/test_plotting.py::test_plotter_scale                         - 13.6 MB
tests/test_plotting.py::test_plot_show_bounds_params[front-both]   - 5.1 MB
tests/test_plotting.py::test_plot_show_bounds_params[all-True]     - 4.0 MB
```

### CODE REVIEW

File: tests/test_filters.py [hunk 1]
```diff
@@ -391,7 +391,7 @@ def test_streamlines():
 
 def test_plot_over_line():
     """this requires matplotlib"""
-    mesh = examples.load_channels()
```
Comment (banesullivan) [inline tests/test_filters.py]: Using this dataset resulted in ~1GB memory consumption for just this test

Comment (banesullivan) [reply tests/test_filters.py]: Using the smaller `load_uniform` now - this should speed up Travis tests

**Comment by banesullivan** [comment]
Memory consumption on travis looks pretty good: https://travis-ci.org/pyvista/pyvista/jobs/566177422#L1128

File: pyvista/plotting/plotting.py [hunk 2]
```diff
@@ -2249,6 +2245,13 @@ def close(self):
             except BaseException:
                 pass
 
+    def deep_clean(self):
+        for renderer in self.renderers:
+            renderer.deep_clean()
+        # Do not remove the renderers on the clean
```
Comment (GuillaumeFavelier) [inline pyvista/plotting/plotting.py]: Why not removing the list of renderers in the `deep_clean` function itself?

File: tests/test_plotting.py [hunk 3]
```diff
@@ -694,3 +694,15 @@ def test_opacity_transfer_functions():
...
     mapping = pyvista.opacity_transfer_function(foo, n)
     assert len(mapping) == n
+
+
+@pytest.mark.skipif(NO_PLOTTING, reason="Requires system to support plotting")
+def test_closing_and_mem_cleanup():
+    n = 5
+    for i in range(n):
+        for j in range(n):
+            p = pyvista.Plotter(off_screen=OFF_SCREEN)
```
Comment (GuillaumeFavelier) [inline tests/test_plotting.py]: I think we should consider using the same test with `BackgroundPlotter` as well in `test_qt_plotting.py`.

**Pull request merged**